### PR TITLE
156 javadoc

### LINKED ISSUE #156: Write Javadoc.

### PR DESCRIPTION
**Pull request by JonatanBonjour**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/ui/UIConstants.java [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package ch.epfl.sdp.ui;
 
+/**
+ * Constants
```
Comment (ksabaa15) [inline app/src/main/java/ch/epfl/sdp/ui/UIConstants.java:4]: ca pourrais etre mieu

File: app/src/main/java/ch/epfl/sdp/ui/event/chat/ChatViewModel.java [hunk 2]
```diff
@@ -17,6 +17,9 @@
 
 import static ch.epfl.sdp.ObjectUtils.verifyNotNull;
 
+/**
+ * View model for the chat
```
Comment (ksabaa15) [inline app/src/main/java/ch/epfl/sdp/ui/event/chat/ChatViewModel.java]: Faudra peut etre dire un petit peu plus de ce qui est deja evident

File: app/src/main/java/ch/epfl/sdp/ui/main/swipe/EventDetailFragment.java [hunk 3]
```diff
@@ -4,25 +4,30 @@
...
-import com.google.firebase.firestore.GeoPoint;
-
 
 import ch.epfl.sdp.Event;
-import ch.epfl.sdp.R;
 import ch.epfl.sdp.databinding.FragmentDefaultEventBinding;
 
+/**
+ * Fragment to display details about an event.
+ * To be replaced !!!
```
Comment (ksabaa15) [inline app/src/main/java/ch/epfl/sdp/ui/main/swipe/EventDetailFragment.java:18]: make it @Deprecated to discourage from using it until removed

**Review by ksabaa15 — APPROVED** [review]
nice

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 4]
```diff
@@ -23,21 +29,45 @@ public Event build() {
...
+     */
     public EventBuilder setDescription(String description) {
         this.mDescription = description;
         return this;
     }
 
+    /**
+     * Set the date attribute for the Event to be created using the Date class
+     *
+     * @param date to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 5]
```diff
@@ -23,21 +29,45 @@ public Event build() {
...
+     */
     public EventBuilder setDate(Date date) {
         this.mDate = date;
         return this;
     }
 
+    /**
+     * Set the date attribute for the Event to be created using human readable date
+     *
+     * @param date to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 6]
```diff
@@ -23,21 +29,45 @@ public Event build() {
         return new Event(mTitle, mDescription, mDate, mAddress, mLocation, mImageId);
     }
 
+    /**
+     * Set the title attribute for the Event to be created
+     *
+     * @param title to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 7]
```diff
@@ -23,21 +29,45 @@ public Event build() {
...
+     */
     public EventBuilder setTitle(String title) {
         this.mTitle = title;
         return this;
     }
 
+    /**
+     * Set the description attribute for the Event to be created
+     *
+     * @param description to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 8]
```diff
@@ -47,16 +77,34 @@ public EventBuilder setDate(String date) {
         return this;
     }
 
+    /**
+     * Set the address attribute for the Event to be created
+     *
+     * @param address to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 9]
```diff
@@ -47,16 +77,34 @@ public EventBuilder setDate(String date) {
...
+     */
     public EventBuilder setAddress(String address) {
         this.mAddress = address;
         return this;
     }
 
+    /**
+     * Set the location attribute for the Event to be created
+     *
+     * @param location to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/EventBuilder.java [hunk 10]
```diff
@@ -47,16 +77,34 @@ public EventBuilder setDate(String date) {
...
+     */
     public EventBuilder setLocation(LatLng location) {
         this.mLocation = location;
         return this;
     }
 
+    /**
+     * Set the image identifier attribute for the Event to be created
+     *
+     * @param imageId to assigned
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/EventBuilder.java]: change `assigned` to `assign`

File: app/src/main/java/ch/epfl/sdp/db/queries/DocumentQuery.java [hunk 11]
```diff
@@ -5,19 +5,67 @@
...
+     * Update a field of the object associated to this query.
+     *
+     * @param field target field to update
+     * @param value object that should replace the targeted field
+     * @param callback class that will be called once updated
+     */
     void update(@NonNull String field, @NonNull Object value, @NonNull OnQueryCompleteCallback<Void> callback);
 
+    /**
+     * Get the livedata the object associated to this query.
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/db/queries/DocumentQuery.java]: there is probably `of` missing between `livedata` and `the`

File: app/src/main/java/ch/epfl/sdp/storage/Storage.java [hunk 12]
```diff
@@ -10,5 +10,11 @@
  * To store an image on firebase storage from an image uri.
  */
 public interface Storage {
+    /**
+     * Method to upload an image on the storage
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/storage/Storage.java]: change `on` to `to`

File: app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java [hunk 13]
```diff
@@ -28,10 +31,20 @@ public AttendingEventViewHolder(@NonNull View itemView) {
         }
     }
 
+    /**
+     * Constructor of the AttendingEventAdapter
+     *
+     * @param attendingEvents a list of events someone attend to
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java]: probably better to change the description of the parameter, I assume this means we get the list of events the user is attending but I am not sure

File: app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java [hunk 14]
```diff
@@ -28,10 +31,20 @@ public AttendingEventViewHolder(@NonNull View itemView) {
...
+     */
     public AttendingEventAdapter(@NonNull List<Event> attendingEvents) {
         mAttendingEvents = verifyNotNull(attendingEvents);
     }
 
+    /**
+     * Method to set a list of events
+     *
+     * @param attendingEvents a list of events someone attend to
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java]: probably better to change the description of the parameter, I assume this means we get the list of events the user is attending but I am not sure

File: app/src/main/java/ch/epfl/sdp/ui/settings/AccountFragment.java [hunk 15]
```diff
@@ -19,6 +17,9 @@
 
 import static ch.epfl.sdp.ObjectUtils.verifyNotNull;
 
+/**
+ * Fragment for the account informations of the user
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/ui/settings/AccountFragment.java]: change `informations` to `information`

File: app/src/main/java/ch/epfl/sdp/ui/main/swipe/EventDetailFragment.java (hunk 3 above)
Comment (JonatanBonjour) [reply app/src/main/java/ch/epfl/sdp/ui/main/swipe/EventDetailFragment.java:18]: ça risque de faire un warning après

File: app/src/main/java/ch/epfl/sdp/ui/UIConstants.java (hunk 1 above)
Comment (JonatanBonjour) [reply app/src/main/java/ch/epfl/sdp/ui/UIConstants.java:4]: j'ai pas d'idées xD

File: app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java [hunk 16]
```diff
@@ -28,10 +31,20 @@ public AttendingEventViewHolder(@NonNull View itemView) {
         }
     }
 
+    /**
+     * Constructor of the AttendingEventAdapter
+     *
+     * @param attendingEvents a list of events the user is attending to
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java]: I think it's better to remove the `to` here

File: app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java [hunk 17]
```diff
@@ -28,10 +31,20 @@ public AttendingEventViewHolder(@NonNull View itemView) {
...
+     */
     public AttendingEventAdapter(@NonNull List<Event> attendingEvents) {
         mAttendingEvents = verifyNotNull(attendingEvents);
     }
 
+    /**
+     * Method to set a list of events
+     *
+     * @param attendingEvents a list of events the user is attending to
```
Comment (dmizr) [inline app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingEventAdapter.java]: I think it's better to remove the `to`

**Review by dmizr — APPROVED** [review]

**Pull request merged**